src/RA_RichPresence.cpp

```diff
@@ -261,17 +281,19 @@ void RA_RichPresenceInterpreter::ParseFromString(const char* sRichPresence)
         }
     }
 
-    if (!sDisplayString.empty())
-    {
-        for (std::vector<std::pair<std::string, std::string>>::const_iterator iter = mDisplayStrings.begin(); iter != mDisplayStrings.end(); ++iter)
-        {
-            auto& displayString = m_vDisplayStrings.emplace_back(iter->first);
-            displayString.InitializeParts(iter->second, mFormats, m_vLookups);
-        }
+    if (sDisplayString.empty())
+        return false;
 
-        auto& displayString = m_vDisplayStrings.emplace_back();
-        displayString.InitializeParts(sDisplayString, mFormats, m_vLookups);
+    for (std::vector<std::pair<std::string, std::string>>::const_iterator iter = mDisplayStrings.begin(); iter != mDisplayStrings.end(); ++iter)
+    {
+        auto& displayString = m_vDisplayStrings.emplace_back(iter->first);
+        displayString.InitializeParts(iter->second, mFormats, m_vLookups);
     }
```

Comment:
This for loop can probably look cleaner as a ranged for. Kind of like this...
```
for(auto& sPair : mDisplayStrings)
{
   auto& displayString = m_vDisplayStrings.emplace_back(sPair.first);
   displayString.InitializeParts(sPair.second, mFormats, m_vLookups);
}
```
The only time I can think of to not use a ranged for loop is if you need rewind or advance the iterator, or if the reference passed isn't used.

Reply:
not new code - indention changed - but will modify as suggested.

Reply:
Sorry I hadn't realized. Thanks anyway.